lib/src/layer/tile_layer/tile_image_view.dart

```diff
@@ -27,24 +27,46 @@ class TileImageView {
           tileImage.loadError && !_visibleRange.contains(tileImage.coordinates))
       .toList();
 
-  List<TileImage> staleTiles() {
-    final tilesInKeepRange = _tileImages.values
-        .where((tileImage) => _keepRange.contains(tileImage.coordinates));
-    final retain = Set<TileImage>.from(tilesInKeepRange);
-
-    for (final tile in tilesInKeepRange) {
-      if (!tile.readyToDisplay) {
-        final coords = tile.coordinates;
-        if (!_retainAncestor(
-            retain, coords.x, coords.y, coords.z, coords.z - 5)) {
-          _retainChildren(retain, coords.x, coords.y, coords.z, coords.z + 2);
+  Iterable<TileImage> staleTiles() {
+    final stale = HashSet<TileImage>();
+    final retain = HashSet<TileImage>();
+
+    for (final tile in _tileImages.values) {
+      final c = tile.coordinates;
+      if (_keepRange.contains(c)) {
+        if (!tile.readyToDisplay) {
+          final retainedAncestor =
+              _retainAncestor(retain, c.x, c.y, c.z, c.z - 5);
+          if (!retainedAncestor) {
+            _retainChildren(retain, c.x, c.y, c.z, c.z + 2);
+          }
         }
+      } else {
+        stale.add(tile);
       }
     }
 
-    return _tileImages.values
-        .where((tileImage) => !retain.contains(tileImage))
-        .toList();
+    return stale.where((tile) => !retain.contains(tile));
+  }
+
+  Iterable<TileImage> renderTiles() {
```

Comment:
This one is a bit more nuanced mostly because TileImageView looks like a public API, which it really isn't. I did document this function, just on the Manager, which seemed like a more public API to me (that's debatable though. I'd rather not duplicate documents, that just creates work. If you want I can move the documentation from the manager to here.

Reply:
I think it's better to keep the documentation inside the manager class for the same reason that you mention.
Maybe you could make the purpose of the method more clear in the name? Like naming it `filterStaleTiles()` or even by using a getter. What do you think? (same would apply to the other filter functions of the model)

Reply:
Are you suggesting to change existing APIs. FWIW, I made renderTiles() match the pre-existing staleTiles(). I'm not partial to any, I'm mostly wondering who the audience is since TimeImageView is an internal API

Reply:
> Are you suggesting to change existing APIs?

Yes. To make it better understandable to someone who dives into the code (code audit, future contributor, ...) that for example `renderTiles()` doesn't mean to render tiles but to filter and return the tiles that should get rendered.

However, if you don't want changes to existing APIs as part of your pull request, I can understand that too. Then leave it as is.

Reply:
> hat for example renderTiles() doesn't mean to render tiles but to filter and return the tiles that should get rendered.

Good point. Render is the wrong verb for  what the method does. Changed